### DISCUSSION TITLE
All input on HoloLens2 triggered 2-3 times after updating to OpenXR plugin.

### DISCUSSION THREAD

**Discussion by leonmueller94** [microsoft/MixedRealityToolkit-Unity#10425] [Q&A]
Hi all,

I'm working on a project for the HoloLens 2 with MRTK 2.7 and Unity. Yesterday, I've updated my Unity version from 2019.4 LTS to 2020.3 LTS. This meant that I had to change to the new OpenXR plugin and XR Plugin Management. 

Most of the things work fine after the update process (hand tracking, rendering, etc.). However, when I deploy and run the app on the HoloLens 2, each select gesture that I perform with the ShellHandRayPointer (i.e. aiming and somthing and pinching my index finger and thumb) triggers twice or three times. For example, if I press on a UI elememt, I can observe that it is being pressed multiple times. As Input Data providers in the MRTK settings I have "OpenXR XRSDK Device Manager", "Input Simulation Service (for running in play mode on PC)", and "Hand Joint Service". 

Any help with this issue is greatly appreciated.

**Comment by polar-kev** [comment]
Hey @leonmueller94, this isn't something we've seen before. Can you let us know 

- if you used the migration tool to do your update?
- if you're also seeing this in the editor/with holographic remoting?
- which types of buttons you're seeing this on and if this also happens in the example scenes?

**Comment by Daniel-Tamborra** [comment]
``We actually solved this issue.

The addiional clicks is due to a "Submit Event" that only happens on HoloLens2 device when OpenXR is the active XR Plug-in provider. It happens exactly "on pinch gesture down"

If you go to your Event System in Unity and disable "Send Navigation Events", this will "solve" your trigger issue. 

This is a workaround and not an actual solution, the reason why it works is due to the fact that UnityEngine.EventSystem not ignores the submit event becuase you told it to not send Navigational Events

The fact remains that OpenXR still sends a Submit Event, which causes problems on all UnityEngine.UI.Button, since it calls the "Press()" function. It is NOT a issue on the MRTK 3D buttons that are used in the example scene, since these are not actually Unity.UI buttons

```
**CLICK STACKTRACE**:
UnityEngine.DebugLogHandler:LogFormat(LogType, Object, String, Object[])
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:LogError(Object)
ClickCounter:OnClick()
UnityEngine.LowMemoryCallback:Invoke()
UnityEngine.Events.InvokableCall:Invoke()
UnityEngine.Events.UnityEvent:Invoke()
UnityEngine.UI.Button:Press()
UnityEngine.UI.Button:OnSubmit(BaseEventData)
UnityEngine.EventSystems.ExecuteEvents:Execute(ISubmitHandler, BaseEventData)
System.AssemblyLoadEventHandler:Invoke(Object, AssemblyLoadEventArgs)
UnityEngine.EventSystems.ExecuteEvents:Execute(GameObject, BaseEventData, EventFunction`1)
UnityEngine.EventSystems.StandaloneInputModule:SendSubmitEventToSelectedObject()
UnityEngine.EventSystems.StandaloneInputModule:Process()
Microsoft.MixedReality.Toolkit.Input.MixedRealityInputModule:Process()
UnityEngine.EventSystems.EventSystem:Update()
```

**Comment by MaxWang-MS** [comment]
@keveleigh, do you believe this is the same issue as #10304?

**Comment by david-c-kline** [comment]
@leonmueller94,

@keveleigh posted a workaround here: https://github.com/microsoft/MixedRealityToolkit-Unity/issues/10304#issuecomment-1040503841

please let us know if this fixes your issue.
